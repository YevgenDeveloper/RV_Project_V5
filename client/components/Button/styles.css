@import 'client/styles/variables';

.button {
  position: relative;
  height: 32px;
  margin: 0;
  padding: 0 20px;
  border-radius: 4px;
  text-align: center;
  font-size: $font-size-default;
  font-weight: bold;
  color: $color-text-default;
  background-position: 50% 50%;
  background-repeat: no-repeat;
  background-color: $color-background;
  transition: background-color 0.1s;

  &::after {
    position: absolute;
    top: 0;
    left: 0;
    opacity: 0.6;
    content: '';
    width: 100%;
    height: 100%;
    background-repeat: no-repeat;
    background-position: 50% 50%;
    transition: opacity 0.1s;
  }

  &:disabled {
    color: $color-text-disabled;
    &::after {
      opacity: 0.2;
    }
  }

  &:hover:enabled {
    cursor: pointer;
    background-color: color-mod($color-background l(90%));
    &::after {
      opacity: 0.8;
    }
  }

  &:active:enabled {
    background-color: color-mod($color-background l(85%));
    &::after {
      opacity: 0.6;
    }
  }

  &.blue {
    color: white;
    background-color: $color-blue;

    &:hover:enabled {
      background-color: color-mod($color-blue l(48%));
    }

    &:active:enabled {
      background-color: color-mod($color-blue l(44%));
    }
  }

  &.dark {
    color: white;
    background-color: #323232;

    &:hover:enabled {
      background-color: #373737;
    }

    &:active:enabled {
      background-color: #2d2d2d;
    }
  }
}
