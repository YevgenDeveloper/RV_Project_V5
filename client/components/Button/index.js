// @flow

import React from 'react';
import classNames from 'classnames/bind';

import styles from './styles.css';

const cx = classNames.bind(styles);

type Props = {
  blue?: boolean,
  dark?: boolean,
  disabled?: boolean,
  label?: string,
  className?: string,
  onClick: any => any,
};

const Button = (props: Props) => (
  <button
    type="button"
    aria-label={props.label}
    title={props.label}
    onClick={props.onClick}
    className={cx(styles.button, props.className, {
      [styles.blue]: props.blue,
      [styles.dark]: props.dark,
    })}
    disabled={props.disabled}
  >
    {props.label}
  </button>
);

Button.defaultProps = {
  label: undefined,
  className: undefined,
  blue: false,
  dark: false,
  disabled: false,
};

export default Button;
