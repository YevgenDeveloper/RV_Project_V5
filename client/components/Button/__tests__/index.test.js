// @flow

import type { ShallowWrapper } from 'enzyme';

import React from 'react';
import { shallow } from 'enzyme';
import Button from '..';

it('renders correctly', () => {
  const wrapper: ShallowWrapper = shallow(<Button />);

  expect(wrapper).toMatchSnapshot();
});
