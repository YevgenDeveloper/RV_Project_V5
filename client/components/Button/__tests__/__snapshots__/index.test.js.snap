// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    blue={false}
    dark={false}
    disabled={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": undefined,
      "children": undefined,
      "className": "button",
      "disabled": false,
      "onClick": undefined,
      "title": undefined,
      "type": "button",
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": undefined,
        "children": undefined,
        "className": "button",
        "disabled": false,
        "onClick": undefined,
        "title": undefined,
        "type": "button",
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
